gateways/gateway.go

```diff
@@ -79,10 +115,211 @@ func (gc *GatewayConfig) newConfigMapWatch(name string) *cache.ListWatch {
 		options.Watch = true
 		options.FieldSelector = fieldSelector.String()
 		req := x.Get().
-			Namespace(gc.Namespace).
+			Namespace(gc.namespace).
 			Resource(resource).
 			VersionedParams(&options, metav1.ParameterCodec)
 		return req.Watch()
 	}
 	return &cache.ListWatch{ListFunc: listFunc, WatchFunc: watchFunc}
 }
+
+// syncs registered configurations and updated gateway configmap
+func (gc *gatewayConfig) manageConfigurations(cm *corev1.ConfigMap) error {
+	newConfigs, err := gc.createInternalConfigs(cm)
+	if err != nil {
+		return err
+	}
+	staleConfigKeys, newConfigKeys := gc.diffConfigurations(newConfigs)
+
+	gc.log.Info().Interface("stale-config-keys", staleConfigKeys).Msg("stale config")
+	gc.log.Info().Interface("new-config-keys", newConfigKeys).Msg("new config")
+
+	// run new configurations
+	for _, newConfigKey := range newConfigKeys {
+		newConfig := newConfigs[newConfigKey]
+		gc.registeredConfigs[newConfigKey] = newConfig
```

Comment:
might want a lock here since despite what we previously discussed. this function is getting executed from `ResourceEventHandlerFuncs` from the watch and we're mutating a shared map.

Reply:
done